examples/casestudies/inference.effekt.md

```diff
@@ -0,0 +1,617 @@
+    do emit(result)
+    val ((result1, trace1), prob1) = handleSample {
+      with handleRejection
+      metropolisStepSingleSite(prob, trace) { trace => 
+        with handleWeight
+        with handleReusingTrace(trace)
+        program()
+      }
+    }
+    loop(result1, trace1, prob1)
+  }
+  loop(result0, trace0, prob0)
+}
+```
+
+## Wrappers
+In order to make it easier to use these algorithms without having to call the various effect handlers, we constructed wrappers for the algorithms implemented in this library.
+
+```
+def sliceSampling[R](n: Int) { program: () => R / { sample, observe } }: Unit / { random, Emit[R] } = {
+  with handleSample
+  with sliceSamplingAlgo[R]
+  with handleObserve
+  program()
+}
+def metropolisHastings[R](n: Int) { program: () => R / { sample, observe } }: Unit / { random, Emit[R] } = {
+  with metropolisHastingsAlgo[R]
+  with handleObserve
+  program()
+}
+def metropolisHastingsSingleSite[R](n: Int) { program: () => R / { sample, observe } }: Unit / { random, Emit[R] } = {
+  with metropolisHastingsSingleSiteAlgo[R]
+  with handleObserve
+  program()
+}
+```
+
+## Examples
+
+### Linear Regression
+As a short example on how the effects `Sample` and `Observe` can be used in an model, we construct the linear regression model.
+
+```
+record Point(x: Double, y: Double)
+
+def show(p: Point): String = {
+  def rounded(d: Double) = round(d, 3).show
+  "Point(" ++ p.x.rounded ++ ", " ++ p.y.rounded ++ ")"
+}
+
+def linearRegression(observations: List[Point]) = {
+  val m = do sample(Gaussian(0.0, 3.0))
+  val c = do sample(Gaussian(0.0, 2.0))
+  observations.foreach {
+    case Point(x,y) => do observe(y, Gaussian((m * x + c), 1.0))
+  }
+  return Point(m, c)
+}
+```
+
+### Robot Movements
+It is also possible to construct bigger examples on which we can apply the algorithms of this library.
+In this example, the movements of a robot on a 2D-plane are observed. In the center of this plane, at the coordinates (0, 0), there is a radar station that can measure the distance to the robot at any point in time.
+The current state of the robot, consisting of position and velocity.
+
+```
+record State(x: Double, y: Double, vx: Double, vy: Double)
+
+def show(s: State): String = {
+  val State(x, y, vx, vy) = s
+  def rounded(d: Double) = round(d, 3).show
+  "State(" ++ x.rounded ++ ", " ++ y.rounded ++ ", " ++ vx.rounded ++ ", " ++ vy.rounded ++ ")"
+}
+```
+
+In this example we also define a type alias for `Measurement`
+
+```
+type Measurement = Double
+// movement of the robot in one time intervall
+def move(s: State): State / sample = {
+  s match {
+    case State(x, y, vx, vy) =>
+      val xa = do sample(Gaussian(0.0, 1.0))
+      val ya = do sample(Gaussian(0.0, 1.0))
+      return State(x+vx+xa, y+vy+ya, vx+xa, vy+ya)
+  }
+}
+// measures distance to robot and observes how probable position is given current state
+def measure(s: State, m: Measurement): Unit / observe = {
```

Comment:
This is a literate file, why have a comment here is this can be part of the text?